src/Aspire.Hosting/DistributedApplicationBuilder.cs

```diff
@@ -345,6 +351,19 @@ public DistributedApplicationBuilder(DistributedApplicationOptions options)
                 _innerBuilder.Services.AddHostedService<ResourceLoggerForwarderService>();
             }
 
+            // Devcontainers & Codespaces
+            _innerBuilder.Services.TryAddEnumerable(ServiceDescriptor.Singleton<IConfigureOptions<CodespacesOptions>, ConfigureCodespacesOptions>());
```

Comment:
Seems odd that this moved?

Reply:
i changed the code here a bit, now i am not registering even `ExecResourceManager` if the mode is not exec.
I really do want to avoid changing all the places where `Operation` is being used, so I made it a bit hacky only for `DistributedApplicationBuilder`. please let me know if you think this is OK